### PR TITLE
Add more Expressions.arrayPath variants

### LINKED ISSUE #1384: Add more Expressions.arrayPath variants

### PR DESCRIPTION
**Pull request by timowest**
Fixes #1384

### CODE REVIEW

**Comment by johnktims** [comment]
This looks okay. I worry that `Expressions` is getting too large but we can discuss that elsewhere.

File: querydsl-core/src/test/java/com/querydsl/core/types/dsl/ExpressionsTest.java [hunk 1]
```diff
@@ -57,6 +57,11 @@ public void  Signature() throws NoSuchMethodException {
                 assertReturnType(Expressions.class.getMethod(type + "Template", Class.class, Template.class, ImmutableList.class));
             }
         }
+
+        // arrays
+        assertReturnType(Expressions.class.getMethod("arrayPath", Class.class, String.class));
+        assertReturnType(Expressions.class.getMethod("arrayPath", Class.class, Path.class, String.class));
+        assertReturnType(Expressions.class.getMethod("arrayPath", Class.class, PathMetadata.class));
```
Comment (johnktims) [inline querydsl-core/src/test/java/com/querydsl/core/types/dsl/ExpressionsTest.java]: Is there a reason to make sure the return type is the same as the name of the method? It just seems like an odd thing to test.

Comment (timowest) [inline querydsl-core/src/test/java/com/querydsl/core/types/dsl/ExpressionsTest.java]: It's a test for the naming convention, the return type simple name is the capitalized method name.

**Comment by timowest** [comment]
@johnktims @Shredder121 Is this ok to merge?

**Comment by johnktims** [comment]
:+1:

**Comment by Shredder121** [comment]
I don't know whether bounded wildcards would be more relaxing, but it looks good.
Maybe we could add a few methods that use them later?
But this PR is good to merge.

**Pull request merged**